src/java/org/apache/cassandra/service/ReplicaFilteringProtection.java

```diff
@@ -340,7 +341,8 @@ public CFMetaData metadata()
             @Override
             public void close()
             {
-                // nothing to do here
+                // If we hit the failure threshold before consuming a single partition, record the current rows cached.
```

Comment:
@adelapena Double-check me here :)

Reply:
I think this records the max cache size once per replica, when we want a single metric registry per query.

I would say that the best place to put the metric saving is the `close` method of the `UnfilteredPartitionIterators.MergeListener` returned by `RFP#mergeController`. However, debug shows that that method is called before creating, using and closing the row merge listener, which is something that I don't understand.

Reply:
Ended up adding a `doOnClose()` method to `PartitionIterators`, which seems to work well.